File: inv3_backend.Solution/Core/Models/Expense.cs
using System.ComponentModel.DataAnnotations;
using Core.Enums;

namespace Core.Models;

public class Expense
{
    public int IdExpense { get; set; }
    [Required]
    public DateTime DateExpense { get; set; }
    public PayMethods PayMethod { get; set; } = PayMethods.card;
    public Provider? Provider { get; set; }
    [Required]
    public decimal? Spent { get; set; }
    public Product? Product { get; set; }
}

File: inv3_app/lib/features/inventory/domain/entities/expense.dart
import 'package:decimal/decimal.dart';
import 'package:equatable/equatable.dart';
import 'package:inv3_app/features/inventory/domain/entities/product.dart';
import 'package:inv3_app/features/inventory/domain/entities/provider.dart';
import 'package:meta/meta.dart';
///
///
///
class Expense extends Equatable
// Equatable helps to value equality such as dart only supports referential equality
{
  final int? id;
  final DateTime? date;
  final String? payMethod;
  final Provider? provider;
  final Decimal? spent;
  final Product? product;

  const Expense({
  @required this.id, 
  @required this.date, 
  @required this.payMethod, 
            this.provider, 
  @required this.spent, 
            this.product
  });

  @override
  List<Object?> get props => [id, date, payMethod, provider, spent, product];
}
